content/docs/rfcs/70/README.md

```diff
@@ -0,0 +1,236 @@
+-   Bob’s signed prekey SigSPKB from X3DH is used as Bob’s initial ratchet public key of the double ratchet.
+
+Once this initialization has been set, Alice and Bob can start exchanging messages with forward secrecy and authentication.
+
+## Specification as a Noise protocol
+
+X3DH has three phases:
+
+1.  Bob publishes his identity key and prekeys to a server, or dedicated smart contract.
+2.  Alice fetches a "prekey bundle" from the server, and uses it to send an initial message to Bob.
+3.  Bob receives and processes Alice's initial message.
+
+One observes that, at the beginning of the protocol, the receiver gets the public key through a server, a smart contract in our situation, together with an encrypted ephemeral key. 
+This corresponds to the Noise pattern  **IX**:
+
+→ e, s \
+← e, s, es, se, ee
+
+The Diffie-Hellman ratchet is run using the valid private key of the receiver in combination with the valid public included in the message coming from the sender. 
+This process is encoded, in Noise terms, as the DH() function. 
+This function will have inputs the secret key of the user running the function, and the public key of the external user. 
+Receiver and sender MUST generate valid key pairs, i.e. points of the X448, using the Noise function GENERATE_KEYPAIR().
+
+The Key Derivation Function (KDF) ratchet and the associated encryption protocols used by the double ratchet are also included by the Noise framework: 
+SHA256 for the KDF and AES256 for AEAD encryption.
+
+Consequently, according to the Noise framework specifications, the X3DH algorithm is encoded as  **Noise_IX_448_AES256GCM_SHA256**
+
+# Retrieving information
+
+## Static data
+
+Some data, such as the key pairs (ik, IK) for Alice and Bob, do not need to be regenerated after a period of time. 
+Therefore the public keys IK can be stored in long-term storage solutions, such as a dedicated smart contract which outputs such a key pair when receiving an Ethereum wallet address.
+
+## Ephemeral data
+
+Storing ephemeral data on Ethereum can be done using a combination of on-chain and off-chain solutions. 
+This approach provides an efficient solution to the problem of storing updatable data in Ethereum.
+1. Ethereum can store a reference or a hash that points to the off-chain data.
+2. Off-chain solutions can include systems like IPFS, traditional cloud storage solutions, or decentralized storage networks such as a [Swarm](https://www.ethswarm.org). 
+In any case, the user stores the associated IPFS hash, URL or reference in Ethereum.
+
+The fact of a user not updating the ephemeral information can be understood as Bob not willing to participate in any communication.
+
+## Interaction with Ethereum
+
+Storing static data is done using a dedicated smart contract *PublicKeyStorage* which associates the Ethereum wallet address of a user with his public key.
```

Comment:
While this what is done with Status, using ENS. I am not convinced this is the only viable solution.

I'd recommend to have a more agnostic approach. Similar to what is done in the "static data" paragraph where several solutions are proposed.